### DISCUSSION TITLE
Random tabs/spaces being inserted and paste-code freezing IDE.

### DISCUSSION THREAD

**Discussion by TheBarret** [community/community#14842] [Copilot Conversations]
I experienced that if you paste code into VS2022 (latest update) it sometimes locks up the IDE.
A modal progress bar pops up stating its looking up symbols, this takes sometimes up to 5secs.
The pasted data might be 1 line or several, in either case it will do this.
When I turn off Copilot, this behavior will stop.

I also experienced that when you start writing or enter a new line, VS/IntelliCode/Copilot inserts a TAB space sometimes.
Haven't seen this before the latest patch.

Anybody else seen/experienced this?

**Comment by kzu** [comment]
Could you share what version you're using? Thanks!

**Reply by TheBarret** [reply]
> Could you share what version you're using? Thanks!
```
Microsoft Visual Studio Community 2022
Version 17.1.4
VisualStudio.17.Release/17.1.4+32407.343
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

Azure App Service Tools v3.0.0   17.1.358.51495
Azure App Service Tools v3.0.0

C# Tools   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitHub Copilot   1.12.0.0 (v1.12.0.0@0660f7705)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.12.5517 (v1.12.0)

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

NuGet Package Manager   6.1.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

OpenOnGitHub   1.3
Visual Studio Extension for opening files on GitHub.com

Syntax Visualizer   1.0
An extension for visualizing Roslyn SyntaxTrees.

Visual Basic Tools   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.21610.4+07b5673e4f2fa7630e78abe37f16b372353a7242
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Inline Suggestions   2.2.1930.216
Ghost text API for Visual Studio inline suggestions

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.
```

**Comment by Mitsuriou** [comment]
I confirm that I have the same issue too. It sometimes (very rare) adds a 2-spaces TAB.
But I cannot say how or when. It is very random, but it happened mostly when starting new lines of code.
I'm using C++.

**Comment by TheBarret** [comment]
It is getting worse since the last patch I applied on VS2022

![](https://i.imgur.com/wQRrCJ1.png)

This happens when I go to into debug (break), it freezes right after that window pops up.

(might be a VS2022 bug, I figured it might be useful to report)

**Reply by kzu** [reply]
That seems entirely unrelated. Can you reproduce that after disabling copilot from the Extensions Manager?

Also, the latest public release should have a fix for the paste hang issue. Could you confirm that's gone now too?

**Reply by TheBarret** [reply]
> That seems entirely unrelated. Can you reproduce that after disabling copilot from the Extensions Manager?
> 
> Also, the latest public release should have a fix for the paste hang issue. Could you confirm that's gone now too?

You are prob right, i did a full re-install of the VS2022 and this has not yet happened again.

**Comment by NathanBarnard** [comment]
I get the same issue when starting a single-line comment (//) on a fresh line. It adds a bunch of whitespace after the slashes.

**Reply by kzu** [reply]
Copilot will auto-format (normalize) the current line to match whatever settings you have for tabs/spaces for the current document language. This is needed for copilot to properly detect open/close brackets and suggest appropriate blocks of code. That said, I can't repro this so perhaps you could record this with some additional context for further  investigation?

![commentline](https://user-images.githubusercontent.com/169707/169840679-b1102870-75a6-4ede-880c-e182bf96aba9.gif)
